File: rb/features.rb
# TempMailApiByBoomlify SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module TempMailApiByBoomlifyFeatures
  def self.make_feature(name)
    case name
    when "base"
      TempMailApiByBoomlifyBaseFeature.new
    when "ratelimit"
      TempMailApiByBoomlifyRatelimitFeature.new
    when "retry"
      TempMailApiByBoomlifyRetryFeature.new
    when "test"
      TempMailApiByBoomlifyTestFeature.new
    when "timeout"
      TempMailApiByBoomlifyTimeoutFeature.new
    else
      TempMailApiByBoomlifyBaseFeature.new
    end
  end
end

File: py/features.py
# TempMailApiByBoomlify SDK feature factory

from feature.base_feature import TempMailApiByBoomlifyBaseFeature
from feature.test_feature import TempMailApiByBoomlifyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TempMailApiByBoomlifyBaseFeature(),
        "test": lambda: TempMailApiByBoomlifyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
